iroh-sync/src/sync.rs

```diff
@@ -100,6 +100,28 @@ pub struct SyncOutcome {
     pub num_sent: usize,
 }
 
+#[derive(Debug, Default)]
+struct Subscribers(Vec<flume::Sender<Event>>);
+impl Subscribers {
+    pub fn subscribe(&mut self, sender: flume::Sender<Event>) {
+        self.0.push(sender)
+    }
+    pub fn unsubscribe(&mut self, sender: &flume::Sender<Event>) {
+        self.0.retain(|s| !s.same_channel(sender));
+    }
+    pub fn send(&mut self, event: Event) {
```

Comment:
shouldn't this only require `&self`?

Reply:
See comment above, the `retain` removes all disconnected senders. This allows to simply drop a receiver without having to explicitly unsubscribe from the replica, because the disconnected sender will be removed on the next emitted event.